src/main/scala/mesosphere/marathon/api/v2/LabelSelector.scala

```diff
@@ -1,16 +1,17 @@
 package mesosphere.marathon.api.v2
 
+import mesosphere.marathon.core.appinfo.AppSelector
 import mesosphere.marathon.state.AppDefinition
 import org.apache.log4j.Logger
 
 import scala.util.control.NonFatal
 import scala.util.parsing.combinator.RegexParsers
 
-case class LabelSelector(key: String, fn: String => Boolean, value: List[String]) {
+case class LabelSelector(key: String, fn: String => Boolean, value: List[String]) extends AppSelector {
```

Comment:
I am not sure. Currently, it is only referenced in the V2 API. It depends on whether we want to use it in future APIs. It does not depend on implementing AppSelector or not (in my opinion).

It should not go into `impl` because it is used outside of the appinfo package. Maybe it even deserves its own package.

Can we make that change part of another PR if you like this?